sympy/solvers/tests/test_solveset.py

```diff
@@ -2310,7 +2206,162 @@ def test_solve_logarithm():
 # end of logarithmic tests
 
 
+# lambert tests
+
+def test_solve_lambert():
+    a = Symbol('a', real=True)
+
+    assert solveset_real(3*log(x) - x*log(3), x) == FiniteSet(
+        3, -3*LambertW(-log(3)/3)/log(3),
+        -3*LambertW(-log(3)/3, -1)/log(3))
+    assert solveset_real(exp(x) - 10*x, x) == FiniteSet(-LambertW(Rational(-1,10)), -LambertW(Rational(-1, 10), -1))
+    assert dumeq(solveset(exp(x) - 10*x, x), ImageSet(Lambda(n, -LambertW(Rational(-1, 10), n)), S.Integers))
+
+    assert solveset_real(exp(x) + x, x) == FiniteSet(-LambertW(1))
+    assert dumeq(solveset(exp(x) + x, x) , ImageSet(Lambda(n, -LambertW(1, n)), S.Integers))
```

Comment:
you may see this one...
even after the symbol is used I make sure to return the dummy...
the point I want to make focus on is that if a `symbol` is used then it can use `subs` or `replace`  in between the functions whereas this can't be done if the dummy is used since we have to reassign the dummy variable in 
another function where it loses the identity and it regards in a different way(obviously this could be done within a function since we don't have to reassign dummy variable).....